### ISSUE TITLE
IDetectionModel に num_classes と model の抽象プロパティを追加

### ISSUE THREAD

**Issue by kurorosu** [kurorosu/pochidetection#466]
## Summary

`IDetectionModel` 抽象クラスに `num_classes` と `model` (内部 `nn.Module`) のプロパティが定義されていない. 各実装 (`RTDetrModel`, `SSD300Model`, `SSDLiteModel`) が個別に `@property` を持つが, インターフェース契約が弱く, 呼び出し側が実装型に依存する状況になっている.

## Background

`IDetectionModel` は SRP / DIP を意識した抽象だが, training / inference の両経路で `num_classes` / `model` アクセスが頻出する. 現状は duck typing で動作しているだけで, mypy や IDE の補完が効かず, 新規モデル追加時に書き忘れが検出されない.

## Current Implementation

- `pochidetection/interfaces/model.py`: `IDetectionModel` に `num_classes` / `model` プロパティなし
- `pochidetection/models/rtdetr.py:139-145`: 実装側で `@property` 定義
- `pochidetection/models/ssd300.py:130-146`: 実装側で `@property` 定義
- `pochidetection/models/ssdlite.py`: 同上

## Proposed Changes

`IDetectionModel` に抽象プロパティを追加:

```python
class IDetectionModel(ABC):
    ...

    @property
    @abstractmethod
    def num_classes(self) -> int:
        """クラス数を返す."""

    @property
    @abstractmethod
    def model(self) -> nn.Module:
        """内部 PyTorch モデルを返す."""
```

- RT-DETR の `processor` は RTDetr 固有のため, 本 Issue では抽象化しない (必要なら別 Issue で `IImageProcessor` を導入検討)
- 既存 3 実装は既にプロパティを持っているため, 実質は interface 側の追加のみで済む

## Acceptance Criteria

- [ ] `IDetectionModel` に `num_classes` / `model` の抽象プロパティを追加する
- [ ] 既存 3 実装の型互換を確認する (追加修正が必要なら反映)
- [ ] `uv run mypy .` が新規エラーを出さない
- [ ] `tests/test_models/` の既存テストが通る
- [ ] training / inference の `num_classes` / `model` 参照箇所で型推論が効く (手動確認)

## Labels

`refactor`

## Branch

`refactor/idetection-model-properties`
